soroban-sdk/test_snapshots/tests/auth/auth_15_one_repeat/test.1.json

```diff
@@ -3,6 +3,54 @@
     "address": 2,
     "nonce": 2
   },
+  "auth": [
+    [
+      "CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFCT4",
```

Comment:
I think it's better than nothing, but OTOH it could be misleading because it's just the last auth. We probably could allow adding a hook for popping the last frame (== finishing a contract call) and then you'd be able to append auths of this particular call to some 'global' auth array. That doesn't need to block this PR though.

Reply:
Hmm, this isn't the auth for all auths that occurred? Is there anyway to get all auths that have occurred?

Reply:
No, we reset auth manager on every host call. This seemed most useful for testing because you don't really want to keep re-asserting on the old entries. We could either additionally require all auths, or add a hook to handle this in more generic fashion (maybe you'd want to report something else per contract call).

Reply:
> on every host call

Do you mean on every top-level `invoke`/`try_invoke` call?

Reply:
I could try persisting the results of calling .auth() in the SDK's `invoke`/`try_invoke` fns. But I don't think the SDK will be able to tell the difference between top-level and nested invoke calls.

Reply:
That's why I'm suggesting to introduce a generic hook in env that would allow executing arbitrary code when we finish a top-level contract invocation.

Reply:
A simple thing to add might be a function `is_in_frame()` or similar that returns true if we're currently inside an invocation. That would be useful in at least one other location in the SDK. Will DM you about how I can add this.

Reply:
@dmkozh and I discussed offline and he highlighted that using a hook would be less error prone for the auth case as there's a few situations where auths can occur and we shoold capture the auths, like when the SDK as_contract fn gets called. We decided to go with the plan outlined in these issues:

- https://github.com/stellar/rs-soroban-env/issues/1218
- https://github.com/stellar/rs-soroban-sdk/issues/1164

And we'd be able to use this functionality to do some other interesting things:
- https://github.com/stellar/rs-soroban-sdk/issues/1165